server/asset/dcr/config.go

```diff
@@ -24,71 +24,95 @@ const (
 	// rpclisten parameter. If the value is an empty string, it will be set
 	// to a default value for the network.
-	DcrdServ string
-	// DcrdServ is the filepath to the dcrd TLS certificate. If it is not
+	RPCListen string `long:"rpclisten" description:"dcrd interface/port for RPC connections (default port: 9109, testnet: 19109)"`
+	// RPCCert is the filepath to the dcrd TLS certificate. If it is not
 	// provided, the default dcrd location will be assumed.
-	DcrdCert string
+	RPCCert string `long:"rpccert" description:"File containing the certificate file"`
 	// Context should be cancelled when the program exits. This will cause some
 	// cleanup to be performed during shutdown.
 	Context context.Context
 }
 
-// tidyConfig initializes and parses the DCRConfig. If there is no error, the
+// loadConfig loads the DCRConfig from file. If no values are found for
+// RPCListen or RPCCert in the specified file, default values will be used.
+// If configPath is an empty string, loadConfig will attempt to read settings
+// directly from the default dcrd.conf filpath. If there is no error, the
 // module-level chainParams variable will be set appropriately for the network.
-func tidyConfig(cfg *DCRConfig) error {
+func loadConfig(configPath string, network asset.Network) (*DCRConfig, error) {
 	// Check for missing credentials. The user and password must be set.
+	cfg := new(DCRConfig)
+
+	// Since we are not reading command-line arguments, and the DCRConfig fields
+	// share names with the dcrd configuration options, passing just
+	// IgnoreUnknown allows us to have the option to read directly from the
+	// dcrd.conf file.
+	parser := flags.NewParser(cfg, flags.IgnoreUnknown)
+
+	// If no path provided, use default dcrd path.
+	if configPath == "" {
+		configPath = defaultConfigPath
+	}
+
+	if _, err := os.Stat(configPath); os.IsNotExist(err) {
+		return nil, fmt.Errorf("no BTC config file found at %s", configPath)
+	} else {
+		// The config file exists, so attempt to parse it.
+		err = flags.NewIniParser(parser).ParseFile(configPath)
+		if err != nil {
+			return nil, fmt.Errorf("error parsing BTC ini file: %v", err)
+		}
+	}
+
 	missing := ""
-	if cfg.DcrdUser == "" {
+	if cfg.RPCUser == "" {
 		missing += " username"
 	}
-	if cfg.DcrdPass == "" {
+	if cfg.RPCPass == "" {
 		missing += " password"
 	}
 	if missing != "" {
-		return fmt.Errorf("missing dcrd credentials:%s", missing)
+		return nil, fmt.Errorf("missing dcrd credentials:%s", missing)
 	}
 
 	// Get network settings. Configuration defaults to mainnet, but unknown
 	// non-empty cfg.Net is an error.
 	var defaultServer string
-	switch cfg.Net {
-	case simnetName:
+	switch network {
+	case asset.Simnet:
 		chainParams = chaincfg.SimNetParams()
 		defaultServer = defaultSimnet
-	case testnetName:
+	case asset.Testnet:
 		chainParams = chaincfg.TestNet3Params()
 		defaultServer = defaultTestnet3
-	case mainnetName, "":
+	case asset.Mainnet:
```

Comment:
I noticed you added regnet in a different spot.  Should we add that here too?

Reply:
`asset.Regtest` is an alias of `asset.Simnet`. 

https://github.com/decred/dcrdex/blob/116de62c773361446b3024d7009f8576f3cf92af/server/asset/common.go#L25

Not all chains are gonna have both, so I figured the DEX will only recognize one network at that level.